include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -892,6 +901,10 @@ struct OrtApi {
   ORT_API2_STATUS(AddSessionConfigEntry, _Inout_ OrtSessionOptions* options,
                   _In_z_ const char* config_key, _In_z_ const char* config_value);
 
+  // Set the languate projection for collecting telemetry data before creating session
+  ORT_API2_STATUS(SetSessionProgrammingProjection, _Inout_ OrtSessionOptions* options,
```

Comment:
i like the idea of a session option, just curious though.   do we expect projetion mix and match?    if not ; we could just make this a global setting, and not something that could change for each session instance.